pkg/mutation/match/match.go

```diff
@@ -22,7 +23,13 @@ const Wildcard = "*"
 // Match selects objects to apply mutations to.
 // +kubebuilder:object:generate=true
 type Match struct {
-	Kinds []Kinds `json:"kinds,omitempty"`
+	// Source determines if the mutator should be used to expand generator
+	// resources. Accepts `Generated`|`Original`|`All` (defaults to `All`). A
```

Comment:
There should be a way to add a regex validation rule via kubebuilder tags so we can validate the user is giving correct input